src/git/zcl_abapgit_git_transport.clas.abap

```diff
@@ -17,6 +17,18 @@ CLASS zcl_abapgit_git_transport DEFINITION
         !eo_branch_list  TYPE REF TO zcl_abapgit_git_branch_list
       RAISING
         zcx_abapgit_exception .
+    CLASS-METHODS upload_pack_by_commit
+      IMPORTING
+        !iv_url          TYPE string
+        !iv_branch_name  TYPE string
```

Comment:
this method is `_by_commit`, so the branch name should not be a input, commits can be related to multiple branches

Reply:
Hmm...that's something I didn't consider: I'd like to get the remote objects of a specific commit within the selected repo branch. So instead of removing the branch name from the signature I would have to integrate it into the git request somehow. Any suggestions?

Reply:
Plus find_branch is private. But the git request could be modified so it'd be able to retrieve specific commits from a specific branch.

Reply:
But it seems I don't need the ev_branch (rs_result-branch) anymore... that's good..so I'll just delete it.. 👍

Reply:
Just for documentation I tried it too and it worked as a fallback: 
```
0068want 82aff720fa970bea0c7281405f0cbbf2bc793c91 refs/heads/master side-band-64k no-progress multi_ack
000Edeepen 99
00000009done
```

Reply:
Resolved with commit 0d17405